services/subscriptions/subscriptions.go

```diff
@@ -0,0 +1,80 @@
+package subscriptions
+
+import (
+	"fmt"
+	"sync"
+	"time"
+)
+
+type Subscriptions struct {
+	mu          sync.Mutex
+	subs        map[SubscriptionID]*Subscription
+	checkPeriod time.Duration
+}
+
+func NewSubscriptions(period time.Duration) *Subscriptions {
+	return &Subscriptions{
+		mu:          sync.Mutex{},
+		subs:        make(map[SubscriptionID]*Subscription),
+		checkPeriod: period,
+	}
+}
+
+func (subs *Subscriptions) Create(namespace string, filter filter) (SubscriptionID, error) {
+	subs.mu.Lock()
+	defer subs.mu.Unlock()
+
+	newSub := NewSubscription(namespace, filter)
```

Comment:
you may as well pass check period to NewSubscription. doesn't look like there is a reason to pass it to Start

Reply:
that's what I did at start, but it isn't used anywhere except in `Start`, so I decided to not use a local var